docs/design.md

```diff
@@ -104,3 +104,14 @@ several ways according to policy:
 
 Then like the usual LibOS pattern it will call the container's entry point as though the container
 was a process unto itself (rather than where it now sits like a library in the LibOS PAL process).
+
+## Local Attestation for Filesystem Decryption Key Provisioning
+
+![EnclaveCC_LA.png](images/EnclaveCC_LA.png)
+
+In most scenarios, it is one-shot transactions used for single filesystem decryption key provisioning. However, the LA protocol can easily be extended to provision multiple keys for more general purposes.
+To achieve this, it is necessary to add a key id to the second request message sent along G_a and app's report, with the ID being used to indicate which key is being requested.
```

Comment:
Yes, it doesn't. I remember a similar issue being raised and discussed at a past weekly enclave-cc meeting, however, we conclude that it won't be added at initial stage as agent enclave has prior knowledge of which key need to be provisioned to app enclave at that phase. @jiazhang0 please chime in if any, Thanks.

Reply:
Thanks! Yep, I understand that the agent enclave needs have prior knowledge - but is this relevant to the proposed protocol (i.e., it's about adding the key ID kind of design instead of the provision steps)? It's just a bit weird that we highlight this in wording but it's not even mentioned in the flow chart. It's also obscure from implementation perspective.

Reply:
> Thanks! Yep, I understand that the agent enclave needs have prior knowledge - but is this relevant to the proposed protocol (i.e., it's about adding the key ID kind of design instead of the provision steps)? It's just a bit weird that we highlight this in wording but it's not even mentioned in the flow chart. It's also obscure from implementation perspective.

Because whether the key id can be automatically enumerated or not is decided by the caller ttrpc service/client and their contexts. At this moment, the caller is not discussed yet. We hope to merge the protocol definition completed prior to the implementation. By then, we could make the final decision to add the key id into protocol or not. @bigdata-memory I suggest to add such a comment to explain why the key id is not mentioned in flow chart. Thanks!

Reply:
I see, thanks for the context info. Make sense to me.